fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/SchedularWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -102,10 +102,10 @@ public void updateSchedulerDetail(final SchedulerDetail schedulerDetail) {
     }
 
     @Override
-    public SchedulerDetail retriveSchedulerDetail() {
+    public SchedulerDetail retrieveSchedulerDetail() {
         SchedulerDetail schedulerDetail = null;
         final List<SchedulerDetail> schedulerDetailList = this.schedulerDetailRepository.findAll();
-        if (schedulerDetailList != null) {
+        if (schedulerDetailList != null && schedulerDetailList.size() > 0) {
             schedulerDetail = schedulerDetailList.get(0);
```

Comment:
If this is the logic, why don't we simple grab the first item via the query? I mean it wasn't you who implemented this but since you're touching it, would be worth to optimize a bit. What do you think?

Reply:
Fixed